### DISCUSSION TITLE
Best Strategy for Measuring a 100A Circuit with 80A EVSE on it via Emporia Vue Local?

### DISCUSSION THREAD

**Discussion by ziptbm** [emporia-vue-local/esphome#130] [Q&A]
Has anyone found a good way to measure circuits greater than 50A reliably with Emporia Vue Local?

I've got a 100A dual-pole circuit with an EVSE on it (80A output).  It's unfortunately the Ford Charge Station Pro, while made by Siemens, has the data only accessible via the FordPass mobile app.  This is one of the many reasons I wanted to add something like an Emporia Vue to my system for better visibility.

Per Emporia support, the 50A sensors are actually capable of reading up to 75A or I could purchase another 200A CT clamp with a 3.5mm to 2.5mm adapter to place on this circuit.  However, since this has a known maximum output and is a circuit for a single device, I'm wondering if there are any other ways I could adjust the HA sensor data to properly reflect what's happening, sort of like the 2x multiplier for dual-pole circuits.   Any suggestions for this scenario?

**Comment by flaviut** [comment]
https://github.com/emporia-vue-local/esphome/discussions/59 <- discussion about different clamps

And you're allowed to multiply by any value you want, not just 2. [ESPHome is super extensible.](https://esphome.io/components/sensor/index.html#multiply)

**Comment by ziptbm** [comment]
After reading more across the discussions, I ended up getting a couple 200A CTs and a 3.5mm to 2.5mm adapter, and tested a bit to see how I could best dial this in to be as close to accurate as possible.  I've tested a few scenarios, but determined I really need to put CTs on both poles of my 100A circuit.  [Per this thread, I decided to try a 200A CT on the C port](https://github.com/emporia-vue-local/esphome/discussions/72), but I've now found an unexpected configuration.

Setup:
- Dual pole 100A Circuit connected to an EVSE, so expecting 80A max output.
- 1x 200A CT with 3.5mm to 2.5mm conversion in side port (understood limit of 75A) for one of the poles - CIRCUIT 1
- 1x 200A CT in Port C for the other pole - CIRCUIT 17

CIRCUIT 1
- 50A CT on a side port with multiplier of 2 on Phase B, I'd max out about 16.5kW (expecting 19.2kW if charging at 80A).
- Changing this 50A CT out for a 200A and upping the multiplier to 8, got me closer to 18kW.  I assume the 75A limitation on the port is impacting this number.

CIRCUIT 17
- 200A CT in Port C on the top with a multiplier of 2 on Phase A.
- I was hoping this would give me a better reading than the side port with 200A CT, but I then discovered I was getting different values between the two sensors.

`      - { phase_id: phase_a, input:  "C", power: { name: "Circuit 17 Power", id: cir17, filters: [ *moving_avg, *pos ] } }
`

COMBINED CIRCUITS - Test 1
- CIRCUIT 1, adjusted multiplier to 4 and still on Phase B
- CIRCUIT 17, removed multiplier and still on Phase A
- create a template sensor for the combination of the two outputs for both power and daily energy.
- CIRCUIT 1 now showed ~9kW and CIRCUIT 17 showed 0kW

COMBINED CIRCUITS - Test 2
- changing CIRCUIT 17 to Phase B made it show readings, which put my total at ~18.6kW in my last test
`      - { phase_id: phase_b, input:  "C", power: { name: "Circuit 17 Power", id: cir17, filters: [ *moving_avg, *pos ] } }
`

So while I now have two sensors that appear to be working close to what's expected, I'm surprised to find them both on Phase B.  And it's also weird that it worked previously on Phase A when there was a multiplier....  Any ideas as to why I should have both poles on Phase B and it should work? Or did I mess something up here?

@flaviut mentioning you because in one of those other answered threads, you suggested using Port C, but indicated you haven't tried it.  I'm wondering if you have any idea why I'm seeing what I describe above.  Thanks!

**Reply by flaviut** [reply]
> I assume the 75A limitation on the port is impacting this number.

There's folks monitoring [1000A circuits.  The current clamps basically turn a current into a voltage signal between 0V and 0.333V.](https://github.com/emporia-vue-local/esphome/discussions/59). The firmware assumes that A-C are 200A, and the 1-16 are 50A, but this is just a convention.

> Changing this 50A CT out for a 200A and upping the multiplier to 8

You need 4x to multiply the assumed top value of 50A to 200A, and then you need another 2x because this is a 240V circuit instead of a 120V circuit.

> I'm surprised to find them both on Phase B. And it's also weird that it worked previously on Phase A when there was a multiplier

[Split phase system](https://en.wikipedia.org/wiki/Split-phase_electric_power), and you are using the `*pos` filter. Without the `*pos` filter, you'd be seeing negative power.

**Reply by ziptbm** [reply]
> > I'm surprised to find them both on Phase B. And it's also weird that it worked previously on Phase A when there was a multiplier
> 
> [Split phase system](https://en.wikipedia.org/wiki/Split-phase_electric_power), and you are using the `*pos` filter. Without the `*pos` filter, you'd be seeing negative power.

Thanks for the response and info.  I'm still not clear on this situation of having both circuits from the dual-pole breaker on the same Phase.  Is this the expected behavior?  I was expecting each line to be on a different phase, which is why I'm surprised to find that I needed to put them both on the same phase...especially after one of the tests worked with them on different phases.

**Reply by flaviut** [reply]
It was late & I was wrong about `*pos`. It actually cuts off negative values.

> both circuits from the dual-pole breaker on the same Phase

Does this help? In a split phase system, the dual-pole breaker provides a 240VAC circuit.

![single-phase-to-split-phase-transformer-schematic-diagram](https://user-images.githubusercontent.com/5213469/221895396-d88e013b-51b7-4e35-ae67-87168cf0ae50.jpg)

You don't need to place a clamp on each side of a dual-pole breaker. There might be some 120V electronics connected to it, but they are probably just a few watts, and you can otherwise get accurate results.

**Reply by ziptbm** [reply]
I understand that 2x 120VAC will provide 240VAC in a dual pole....I just don't understand why in the configuration yaml I have both CTs listed on the same phase to get readings.  I would have thought one of the poles would be on A and the other on B.

As for using one CT to get readings from both, I tried that first, but when using a 50A CT, I wasn't getting the full readings due to it being a 100A Circuit with 80A output.  Note the ONLY item on this circuit is the EVSE attached to it.   

And as I outlined above, I did find that I'm getting different readings from each side of the dual-pole breaker on this circuit...although, now that I look at it again, I'm noticing the following differences with 200A CTs on each leg:
- Charging - <500W
- Not charging - <15W


Just to confirm, if I put CTs on both poles of this circuit, it should be expected that I put both line items in the config to the same Phase (B in this case).  Correct?

**Reply by flaviut** [reply]
Sorry, I'm having a lot of trouble wrapping my head around what's going on here.

I don't think I can really help, but I can suggest that removing the `*pos` makes the data clearer in some cases, with the drawback of a little more noise.

**Reply by ziptbm** [reply]
> Sorry, I'm having a lot of trouble wrapping my head around what's going on here.

I'm sorry if I'm explaining this poorly....this is all new to me and there's a bunch here that I'm still learning.

When you attach a CT to a circuit in your panel (in a 2 phase system), you need to assign the "phase_id" to either "phase_a" or "phase_b".   If "phase_a" shows 0w, then switching it to "phase_b" most likely will report power usage.

**My Question - If you have a 2 Phase system and use 2 CT sensors on a single dual-pole circuit, with one CT on each leg, should the YAML configuration have the same phase_id for both sensors or should the YAML configuration have different phase_ids for each sensor?** 

Since the Phases of the circuit alternate every row of the panel ([as outlined in the visual here](https://github.com/emporia-vue-local/esphome#panel-installation-part-1), it would logically make sense that a dual-pole circuit would physically sit on both phases, so it would have one leg on Phase A and the other on Phase B.

However, what I've found in my case is that both of the lines for my 100A dual-pole circuit only report data via Emporia Vue Local when the YAML configuration is set to "phase_b" on both sensors.  Here is the code from my YAML configuration for these two 200A CT clamps, which appears to be working....I'm just surprised both are on Phase B, which is why I'm trying to understand why.


```
- { phase_id: phase_b, input:  "1", power: { name:  "Circuit 1 Power", id:  cir1, filters: [ *moving_avg, *pos, multiply: 4 ] } }    
- { phase_id: phase_b, input:  "C", power: { name: "Circuit 17 Power", id: cir17, filters: [ *moving_avg, *pos ] } }

```

**Reply by flaviut** [reply]
You're being very clear--I just don't know. If the data looks correct to you the way you have it, I would call it good.

Hopefully someone with a better understanding can reply with a principled answer.

**Reply by cdalessio** [reply]
US household wiring is single phase 240v center tapped to provide two 120v 180° out of phase; when measuring 240v I'd only expect one phase, which seems to be the observation here as well.
